src/tiled/mainwindow.cpp

```diff
@@ -498,11 +499,12 @@ void MainWindow::commitData(QSessionManager &manager)
 
 void MainWindow::closeEvent(QCloseEvent *event)
 {
-    writeSettings();
-
-    if (confirmAllSave())
+    if (confirmAllSave()) {
+        //Make sure user won't end up in view mode on next launch
+        toggleClearView(false);
+        writeSettings();
         event->accept();
-    else
+    } else
```

Comment:
Coding style: Please also wrap the else part with brackets. The general rule is that when any part of a condition has curly brackets, all parts should have them.